### ISSUE TITLE
enhancement request: Support preseeding the new KMS provider

### ISSUE THREAD

**Issue by jalaziz** [localstack/localstack#8910]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Enhancement description

It would be great if the new KMS provider could be pre-seeded with actual key material.

The former `local-kms` provider support pre-seeding, but also lacked several endpoints. While new provider recently [added support for custom key IDs](https://github.com/localstack/localstack/commit/82bc739e2286c2a29bfcdf9c4e6ad3856adb9694), there are scenarios where this is not sufficient.

In particular, when developing against web3 smart contracts, having static key material for testing can be necessary to establish trust between local services and the blockchain.

Unfortunately, using `local-kms` isn't an option due to the lack of HMAC support.

### 🧑‍💻 Implementation

Ideally the key material could be loaded from a local file mounted into the localstack container, similar to how `local-kms` worked. Another option could be an API that can be called via an initialization hook.

**Comment by clemstoquart** [comment]
Hello,

Is there any plan to support it ? 
Since `local-kms` isn't available anymore this prevents us to upgrade to localstack 3+.

**Comment by edhgoose** [comment]
We'd love to have this feature too.

Our use case is we use Localstack for our local environments. We store some secrets in a local database, and when someone reboots their computer, Localstack gets reset, which then means the secrets they've stored are invalidated.

We used to be able to seed KMS, which meant the keys were consistent and the values of those keys too. 

Is there a way around this?

**Comment by alechirsch** [comment]
Same here, this feature is crucial to my team being able to upgrade.

**Comment by jalaziz** [comment]
I've pushed a PR that enables support for symmetric (and HMAC) KMS Keys. Asymmetric KMS keys are a bit of a different beast but may be possible.

It uses the custom tag based approach that is already used for custom Key IDs.

**Comment by HarshBalyan** [comment]
@jalaziz can you direct me to some notes for how to setup KMS seeds after your change?

**Comment by jalaziz** [comment]
@HarshBalyan the change still has not been released, but you can setup KMS seeds by using the AWS CLI or SDK to create a KMS key with specific tags.

For example, you can use [initialization hooks](https://docs.localstack.cloud/references/init-hooks/) and run the following command:
```
awslocal kms create-key --tags TagKey=_custom_id_,TagValue=<seed-key-id> TagKey=_custom_key_material_,TagValue=<base64-encoded-key-seed>
```

**Comment by alechirsch** [comment]
I saw that this change was released. The above suggestion from @jalaziz does not seem to be working. Is there any documentation for seeding kms keys? We need to be able to encrypt data, reboot localstack, and be able to decrypt that previously encrypted data.

**Comment by jalaziz** [comment]
> I saw that this change was released. The above suggestion from @jalaziz does not seem to be working. Is there any documentation for seeding kms keys? We need to be able to encrypt data, reboot localstack, and be able to decrypt that previously encrypted data.

This is what I'm using and it works fine (I don't need a static key ID):
```
keyId=$(awslocal kms create-key --key-usage GENERATE_VERIFY_MAC --key-spec HMAC_256 --tags TagKey=_custom_key_material_,TagValue="${KMS_KEY_SEED}" --output text --query "KeyMetadata.KeyId")
awslocal kms create-alias --alias-name alias/foo --target-key-id $keyId
```

`KMS_KEY_SEED` is a base64-encoded string of the key material.

**Comment by alechirsch** [comment]
@jalaziz  All is well, we got it working after reviewing more.

**Comment by edhgoose** [comment]
Thank you for doing this @jalaziz - I think I'm missing something / a little puzzled. Could you offer some guidance if we're doing things right?

```
awslocal kms create-key --region=eu-west-1 --tags \
    TagKey=_custom_key_material_,TagValue="420684aeabda8091243498e0fc27bd3f" TagKey=_custom_id_,TagValue="e39b12ba-c0c1-46de-a511-0820e0574217"
```

Then we're encrypting the word "test":

```
awslocal kms encrypt --region=eu-west-1 --plaintext test --key-id e39b12ba-c0c1-46de-a511-0820e0574217
```

Which gives the following result:

```
{
    "CiphertextBlob": "ZTM5YjEyYmEtYzBjMS00NmRlLWE1MTEtMDgyMGUwNTc0MjE3XJQme4KElMasMn3N0dqVWeaOR77mfa5enCUY3+FQTLNSso/wCooSDWus+endb0Q+",
    "KeyId": "arn:aws:kms:eu-west-1:000000000000:key/e39b12ba-c0c1-46de-a511-0820e0574217"
}
```

All good so far - and I can decrypt that too:

```
awslocal kms decrypt --region=eu-west-1 --ciphertext-blob fileb://<(echo "ZTM5YjEyYmEtYzBjMS00NmRlLWE1MTEtMDgyMGUwNTc0MjE3XJQme4KElMasMn3N0dqVWeaOR77mfa5enCUY3+FQTLNSso/wCooSDWus+endb0Q+" | base64 -d) --key-id "e39b12ba-c0c1-46de-a511-0820e0574217"
```

gives:
```
{
    "KeyId": "arn:aws:kms:eu-west-1:000000000000:key/e39b12ba-c0c1-46de-a511-0820e0574217",
    "Plaintext": "dGVzdA=="
}
```

All as I'd expect.

But if I stop Localstack, and then boot it back up again, and execute that first line to create the key again... the decrypt process no longer works.

I would have expected the same key material to be able to decrypt again. 

Am I missing something? Any help you can offer would be most appreciated. Thanks!!

**Comment by jalaziz** [comment]
> Am I missing something? Any help you can offer would be most appreciated. Thanks!!

I just tested this locally and I can't reproduce your issue. However, I also can't decrypt the ciphertext payload you provided using the key material in your example (which technically I should be able to).

Are you sure you're using the correct version of localstack? This is only available in localstack 3.3.0 and later.
